infrastructure/src/main/java/org/corfudb/infrastructure/LogReplicationServer.java

```diff
@@ -0,0 +1,114 @@
+package org.corfudb.infrastructure;
+
+import io.netty.channel.ChannelHandlerContext;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.logreplication.LogReplicationConfig;
+import org.corfudb.infrastructure.logreplication.receive.LogReplicationSinkManager;
+import org.corfudb.protocols.wireprotocol.CorfuMsg;
+import org.corfudb.protocols.wireprotocol.CorfuMsgType;
+import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
+import org.corfudb.protocols.wireprotocol.logreplication.LogReplicationEntry;
+import org.corfudb.protocols.wireprotocol.logreplication.LogReplicationNegotiationResponse;
+import org.corfudb.protocols.wireprotocol.logreplication.LogReplicationQueryLeaderShipResponse;
+import org.corfudb.protocols.wireprotocol.logreplication.MessageType;
+
+import javax.annotation.Nonnull;
+import java.lang.invoke.MethodHandles;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * This class represents the Log Replication Server, which is
+ * responsible of providing Log Replication across sites.
+ *
+ * The Log Replication Server, handles log replication entries--which
+ * represent parts of a Snapshot (full) sync or a Log Entry (delta) sync
+ * and also handles negotiation messages, which allows the Source Replicator
+ * to get a view of the last synchronized point at the remote site.
+ */
+@Slf4j
+public class LogReplicationServer extends AbstractServer {
+
+    private static final String configFilePath = "/config/corfu/log_replication_config.properties";
+
+    private final ServerContext serverContext;
+
+    private final ExecutorService executor;
+
+    @Getter
+    private final LogReplicationSinkManager sinkManager;
+
+    @Getter
+    private final HandlerMethods handler = HandlerMethods.generateHandler(MethodHandles.lookup(), this);
+
+    public LogReplicationServer(@Nonnull ServerContext context) {
+        this.serverContext = context;
+        this.executor = Executors.newFixedThreadPool(1,
+                new ServerThreadFactory("LogReplicationServer-", new ServerThreadFactory.ExceptionHandler()));
```

Comment:
can make this a single thread executor - Executors.newSingleThreadExecutor(..)

Reply:
We should replace 1 with a passing in value. It is not always 1.